format/spec.md

```diff
@@ -444,28 +456,28 @@ The schema of a manifest file is a struct called `manifest_entry` with the follo
+| _optional_ |            | ~~**`107  sort_columns`**~~       | `list<112: int>`             | **Deprecated. Do not write.**                                                                                                                                                                                                                                                                  |
+| _optional_ | _optional_ | **`108  column_sizes`**           | `map<117: int, 118: long>`   | Map from column id to the total size on disk of all regions that store the column. Does not include bytes necessary to read other columns, like footers. Leave null for row-oriented formats (Avro)                                                                                            |
+| _optional_ | _optional_ | **`109  value_counts`**           | `map<119: int, 120: long>`   | Map from column id to number of values in the column (including null and NaN values)                                                                                                                                                                                                           |
+| _optional_ | _optional_ | **`110  null_value_counts`**      | `map<121: int, 122: long>`   | Map from column id to number of null values in the column                                                                                                                                                                                                                                      |
+| _optional_ | _optional_ | **`137  nan_value_counts`**       | `map<138: int, 139: long>`   | Map from column id to number of NaN values in the column                                                                                                                                                                                                                                       |
+| _optional_ | _optional_ | **`111  distinct_counts`**        | `map<123: int, 124: long>`   | Map from column id to number of distinct values in the column; distinct counts must be derived using values in the file by counting or using sketches, but not using methods like merging existing distinct counts                                                                             |
+| _optional_ | _optional_ | **`125  lower_bounds`**           | `map<126: int, 127: binary>` | Map from column id to lower bound in the column serialized as binary [1]. Each value must be less than or equal to all non-null, non-NaN values in the column for the file [2] For Geometry type, this is a Point composed of the min value of each dimension in all Points in the Geometry.   |
```

Comment:
How does this work? Does Iceberg need to interpret each WKB to produce this value? Will it be provided by Parquet?

Reply:
Yes, once we switch to Geometry logical type from Parquet we will get these stats.

Reply:
Should we mention that it is the parquet type `BoundingBox`?

Reply:
yea will add a footnote here

Reply:
@szehon-ho BTW, the reason why We had a separate bbox statistics in havasu is to be compatible with existing Iceberg tables. Since this is to add the native geometry support, so `lower_bound` and `upper_bound` are good choices.

Reply:
Does it support different dimensions like XY, XYZ, XYM, XYZM? If yes, how can we tell if the binary is for XYZ or XYM?

Reply:
Yea I think z, m are just 0 in this case?

Reply:
I thought that the bounds were stored as WKB-encoded points (according to Appendix D and G), and WKB encodes dimensions of geometries in the header. Making bound values the same type/representation as the field data type is more consistent.

[More sophisticated coverings](https://github.com/apache/parquet-format/pull/240/files#diff-834c5a8d91719350b20995ad99d1cb6d8d68332b9ac35694f40e375bdb2d3e7cR257-R269) in Parquet statistics cannot be easily mapped to `lower_bounds` and `upper_bounds`, so do we simply use the `bbox` statistics and ignore the `coverings` for now?

Reply:
I thought that the bounds were stored as WKB-encoded points (according to Appendix D and G), and WKB encodes dimensions of geometries in the header. Making bound values the same type/representation as the field data type is more consistent.

[More sophisticated coverings](https://github.com/apache/parquet-format/pull/240/files#diff-834c5a8d91719350b20995ad99d1cb6d8d68332b9ac35694f40e375bdb2d3e7cR257-R269) in Parquet statistics cannot be easily mapped to `lower_bounds` and `upper_bounds`, so do we simply use the `bbox` statistics and ignore the `coverings` for now?

Reply:
I think it is ok since these two bounds are optional and in case they are not presented, it still follow the spec.

Reply:
Done, thanks.

Reply:
@flyrain 
> Should we mention that it is the parquet type BoundingBox?

Actually looking again after some time, not sure how to mention this here, as that is filetype specific.  This is an optional field, and only set if type is parquet and bounding_box is set, but that's implementation detail .